modules/store/spec/store.spec.ts

```diff
@@ -471,4 +476,174 @@ describe('ngRx Store', () => {
       mockStore.dispatch(action);
     });
   });
+
+  describe('Meta Reducers', () => {
+    let metaReducerContainer: any;
+
+    beforeEach(() => {
+      metaReducerContainer = (function() {
+        function metaReducer1(reducer: ActionReducer<any, any>) {
+          return function(state: any, action: Action) {
+            return reducer(state, action);
+          };
+        }
+
+        function metaReducer2(reducer: ActionReducer<any, any>) {
+          return function(state: any, action: Action) {
+            return reducer(state, action);
+          };
+        }
+
+        return {
+          metaReducer1: metaReducer1,
+          metaReducer2: metaReducer2,
+        };
+      })();
+    });
+
+    it('should create a meta reducer for root and call it through', () => {
+      const metaReducerSpy: any = spyOn(
+        metaReducerContainer,
+        'metaReducer1'
+      ).and.callThrough();
+      setup({ counter1: 0, counter2: 1 }, [metaReducerContainer.metaReducer1]);
+      const action = { type: INCREMENT };
+      store.dispatch(action);
+      expect(metaReducerSpy).toHaveBeenCalled();
+    });
+
+    it('should call two meta reducers', () => {
+      const metaReducerSpy1: any = spyOn(
+        metaReducerContainer,
+        'metaReducer1'
+      ).and.callThrough();
+      const metaReducerSpy2: any = spyOn(
+        metaReducerContainer,
+        'metaReducer2'
+      ).and.callThrough();
+
+      setup({}, [
+        metaReducerContainer.metaReducer1,
+        metaReducerContainer.metaReducer2,
+      ]);
+      const action = { type: INCREMENT };
+      store.dispatch(action);
+
+      expect(metaReducerSpy1).toHaveBeenCalled();
+      expect(metaReducerSpy2).toHaveBeenCalled();
+    });
+
+    it('should create a meta reducer for feature and call it with the expected reducer', () => {
+      const metaReducerSpy1: any = spyOn(
+        metaReducerContainer,
+        'metaReducer1'
+      ).and.callThrough();
+      const metaReducerSpy2: any = spyOn(
+        metaReducerContainer,
+        'metaReducer2'
+      ).and.callThrough();
+
+      TestBed.configureTestingModule({
+        imports: [
+          StoreModule.forRoot({}),
+          StoreModule.forFeature('counter1', counterReducer, {
+            metaReducers: [metaReducerContainer.metaReducer2],
+          }),
+          StoreModule.forFeature('counter2', counterReducer, {
+            metaReducers: [metaReducerContainer.metaReducer1],
+          }),
+        ],
+      });
+      const mockStore = TestBed.get(Store);
+      const action = { type: INCREMENT };
+      mockStore.dispatch(action);
+
+      expect(metaReducerSpy1).toHaveBeenCalledWith(counterReducer);
+      expect(metaReducerSpy2).toHaveBeenCalledWith(counterReducer);
```

Comment:
What about creating another `counterReducer` and check if the right meta-reducer is called with the correct reducer?